src/model/writer.js

```diff
@@ -150,6 +150,10 @@ export default class Writer {
 	 * second parameter is a {@link module:engine/model/item~Item model item}.
 	 */
 	insert( item, itemOrPosition, offset ) {
+		if ( this.model._currentWriter !== this ) {
+			throw new CKEditorError( 'writer-detached-writer-tries-to-modify-model: Detached writer tries to modify the model' );
+		}
```

Comment:
Wouldn't it be better to have it in a private method?

Reply:
Probably yes. I moved it to the separate private method.